webapp/app/js/directives/clonejobs.js

```diff
@@ -93,6 +93,96 @@ treeherder.directive('thCloneJobs', [
     });
 
     $rootScope.$on(
+        thEvents.selectNextJob, function(ev){
+
+        var jobMap = ThResultSetModel.getJobMap($rootScope.repoName);
+        var lastJobSelected = ThResultSetModel.getSelectedJob($rootScope.repoName);
+        var el, key, job, step;
+        var step = 'nextElementSibling'
+        
+        if(lastJobSelected.el == undefined || lastJobSelected.el[0] == undefined ){
+
+            var currentElement = document.getElementsByClassName('result-set')[0];
+            el = traceJob($(currentElement),step,'job-btn');
+
+        }else{
+
+            var currentElement = lastJobSelected.el[0];    
+
+            //Parsing for parent which satisy parameter mentioned in variable step
+            while(!currentElement[step] || $(currentElement[step]).css('display') == 'none'){
+
+                if(!currentElement[step]){
+                    currentElement = traceParent($(currentElement), 1)[0];
+                }else{
+                    currentElement = currentElement[step];            
+                }
+
+            } 
+
+            //look through parent's sibling for element with class job-row
+            el = traceJob($(currentElement[step]),step,'job-btn');
+
+        }
+        
+        
+        
+        
+        if(el != null){
+
+            key = el.attr(jobKeyAttr);
+            job = jobMap[key].job_obj;
+            selectJob(job);    
+
+        }
+    });
+
+    $rootScope.$on(
+        thEvents.selectPreviousJob, function(ev){
+        
+        var jobMap = ThResultSetModel.getJobMap($rootScope.repoName);
+        var lastJobSelected = ThResultSetModel.getSelectedJob($rootScope.repoName);
+        var el, key, job, jobsList, i, step, currentElement;
+
+        step = 'previousElementSibling';
+        el = null;
+
+        if(lastJobSelected.el === undefined || lastJobSelected.el[0] === undefined){
+            
+            var resultsets = document.getElementsByClassName('result-set');
+            currentElement = resultsets[resultsets.length - 1];
+            el = traceJob($(currentElement),step,'job-btn');
+
+        }else{
+            
+            currentElement = lastJobSelected.el[0];
+
+            while(!currentElement[step] ){
+                currentElement = traceParent($(currentElement), 1)[0];    
+            }   
+
+            el = traceJob($(currentElement[step]),step,'job-btn');                
+
+        }  
+        
+        while(el == null){
+
+            currentElement = traceParent($(currentElement),1)[0]; 
+            if(currentElement === undefined)
+                break;
+            el = traceJob($(currentElement[step]),step,'job-btn');    
+
+        }
+        if(el != null){
+            key = el.attr(jobKeyAttr);
+            job = jobMap[key].job_obj;
+            selectJob(job);         
+        }
+        
+
+
+    });
```

Comment:
I think we can add a new line below this semicolon, and remove the empty lines above to group this with the function block it belongs to.